distribution/helm/vector-agent/values.yaml

```diff
@@ -6,6 +6,11 @@
 # commonLabels:
 #   app.kubernetes.io/component: agent
 
+secrets: {}
+  # aws:
+  #  accessKeyId: ""
+  #  secretAccessKey: ""
```

Comment:
Predefined today - we could flip it and require users to supply the `base64` encoded value, rather than the raw values - which would allow for arbitrary secrets.

Arbitrary secrets could also be tied into auto-generating the config to include them as envvars (through `range` functions)

Reply:
🤔 either way I need to document how the value should be provided (`base64` vs raw) - I just don't want people to think a `base64` encoded value is "secure" to store in git here.

Reply:
FWIW, I tagged the original requester to get their thoughts as well

Reply:
@StephenWakely's also leaning towards arbitrary keys here.

I think the biggest challenge on arbitrary keys is generating appropriate env var names, ex: [aws](https://vector.dev/docs/reference/configuration/sources/aws_s3/#aws_access_key_id) looks for a specific name here - we should be able to pipe the value through `upper` and `snakecase`, with some documentation for the user. Or just make the user setup their own env vars (through that already exposed option)